filters/tee/tee.go

```diff
@@ -263,16 +364,21 @@ func (spec *teeSpec) CreateFilter(config []interface{}) (filters.Filter, error)
 		rx, err := regexp.Compile(expr)
 
 		if err != nil {
+			client.Close()
 			return nil, err
 		}
 		tee.typ = pathModified
 		tee.rx = rx
 		tee.replacement = replacement
 
 		return &tee, nil
+	default:
+		teeClients.mu.Lock()
+		delete(teeClients.store, u.Host)
+		teeClients.mu.Unlock()
+		client.Close()
+		return nil, filters.ErrInvalidFilterParameters
```

Comment:
I think this is not correct as it may close a client used by another filter instance

Reply:
fixed